src/signer.rs

```diff
@@ -226,6 +227,25 @@ impl LightsparkSigner {
         Ok(self.build_commitment_secret(commitment_seed, per_commitment_point_idx))
     }
 
+    pub fn get_payment_preimage_nonce(&self) -> u64 {
```

Comment:
Will this be used anywhere? Isn't sparkcore generating the nonce?

Reply:
Client should generate the nonce, and then pass to sparkcore in the create_invoice mutation.